### PR TITLE
[2.8][DependencyInjection] improve deprecation messages

### PR DESCRIPTION
**Pull request by xabbuh**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets |  |
| License | MIT |
| Doc PR |  |

Also include the service id in the deprecation message.

### CODE REVIEW

**Comment by dosten** [comment]
IMO we should do the same with the `synchronized` tag. What do you think?

**Comment by dosten** [comment]
:+1:

**Comment by nicolas-grekas** [comment]
:+1:

File: src/Symfony/Component/DependencyInjection/Loader/XmlFileLoader.php [hunk 1]
```diff
@@ -162,7 +162,7 @@ private function parseDefinition(\DOMElement $service, $file)
             $triggerDeprecation = 'request' !== (string) $service->getAttribute('id');
 
             if ($triggerDeprecation) {
-                @trigger_error(sprintf('The "scope" attribute in file "%s" is deprecated since version 2.8 and will be removed in 3.0.', $file), E_USER_DEPRECATED);
+                @trigger_error(sprintf('The "scope" attribute of service "%s" in file "%s" is deprecated since version 2.8 and will be removed in 3.0.', (string) $service->getAttribute('id'), $file), E_USER_DEPRECATED);
```
Comment (nicolas-grekas) [inline src/Symfony/Component/DependencyInjection/Loader/XmlFileLoader.php]: why casting to string?

**Comment by fabpot** [comment]
Thank you @xabbuh.

**Pull request merged**